### ISSUE TITLE
Consider removing chpl_task_executeTasksInList()

### ISSUE THREAD

**Issue by ronawho** [chapel-lang/chapel#7223]
`chpl_task_executeTasksInList()` is meant to allow a task spawning coforalls/cobegins to help process the task queue. i.e. for something like `coforall i in 1..10` the tasking layer only really needs to create 9 tasks, and the original task can do the last chunk. 

Currently only fifo actually uses `chpl_task_executeTasksInList()`, and as of https://github.com/chapel-lang/chapel/pull/7222, we don't need it anymore. 

We could consider removing `chpl_task_executeTasksInList()`, which would slightly simply the tasking layer interface and allow us to dramatically simplify fifo tasking.

This is a really low priority item because fifo seems to be working well as-is and even though this would simplify it, it'd still be additional work.

Currently getting rid of `chpl_task_executeTasksInList()` causes timeouts for a few tests (notably ones that set `CHPL_RT_NUM_THREADS_PER_LOCALE=1` or ones that test the deadlock reporting): 

    parallel/cobegin/hilde/reports
    parallel/cobegin/stonea/reports
    parallel/coforall/stonea/reports
    parallel/taskPool/figueroa/OneThread
    parallel/taskPool/figueroa/OneThreadSyncStmt
    parallel/taskPool/figueroa/QueuedTasks

**Comment by gbtitus** [comment]
Right, all the tasks=fifo logic around having the thread that was running the parent task of a parallel construct make itself available to run the children of that construct was done in order to avoid starvation deadlock in situations with limited threads.  And then of course the structure of that logic drove the architecture of the interface, because fifo was the first tasking layer implementation.  (Any layer that can switch tasks on threads doesn't have this problem, which is why executeTasksInList is empty for tasks=qthreads and massivethreads, and also was for muxed.  If we're willing to say that we don't need to run in limited-threads situations like that, then we can simplify both the interface and the fifo implementation.  I think this would be fine.
